src/Utilities/Compiler/WellKnownTypeNames.cs

```diff
@@ -296,16 +296,16 @@ internal static class WellKnownTypeNames
         public const string SystemThreadingInterlocked = "System.Threading.Interlocked";
         public const string SystemThreadingMonitor = "System.Threading.Monitor";
         public const string SystemThreadingSpinLock = "System.Threading.SpinLock";
-        public const string SystemThreadingTasksTask1 = "System.Threading.Tasks.Task`1";
+        public const string SystemThreadingTasksGenericTaskCompletionSource1 = "System.Threading.Tasks.TaskCompletionSource`1";
```

Comment:
```suggestion
        public const string SystemThreadingTasksTaskCompletionSource1 = "System.Threading.Tasks.TaskCompletionSource`1";
```

Reply:
Oh crap... Sorry for the wasted time!

Reply:
No worries, @Evangelink ! :smiley: Sorry I didn't mention to also rename the references.